tensorboard/backend/application.py

```diff
@@ -345,9 +365,14 @@ class are WSGI applications.
     clean_path = _clean_path(parsed_url.path, self._path_prefix)
 
     # pylint: disable=too-many-function-args
-    if clean_path in self.data_applications:
-      return self.data_applications[clean_path](environ, start_response)
+    if clean_path in self.exact_routes:
+      return self.exact_routes[clean_path](environ, start_response)
     else:
+      for path_prefix in self.prefix_routes:
```

Comment:
When I read this block, I look up to see how `self.prefix_routes` was
initialized: it’s initialized to `{}`, an unordered `dict`, so it’s not
clear that this `for`-loop’s iteration order is in fact well defined
because the attribute is later converted to an `OrderedDict`.

Maybe consider refactoring to populate a `raw_prefix_routes` local
variable, or at least add a comment at the original initialization of
`self.prefix_routes` to indicate that its type changes at runtime and
what the contract is?

Reply:
done via unordered_prefix_routes.